### PR TITLE
Implement vacuum index operation

### PR DESCRIPTION
**Pull request by dai-chen**
### Description

Implemented the `VACUUM` index statement in the plugin. The semantics slightly differ from what's defined in Flint Spark: similar to the current `DROP` index logic in the plugin, the vacuum operation can transition from `REFRESHING`, `ACTIVE`, or `DELETED` states directly, instead of only accepting `DELETED`.
 
### Issues Resolved

https://github.com/opensearch-project/opensearch-spark/issues/104
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/build.gradle [hunk 1]
```diff
@@ -111,6 +111,7 @@ configurations.all {
     resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.9.10"
     resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.9.10"
     resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-common:1.9.10"
+    resolutionStrategy.force "net.bytebuddy:byte-buddy:1.14.9"
```
Comment (dai-chen) [inline plugin/build.gradle:114]: OS core bumps Jackson version which in turn bumps byte-buddy version to 1.14.9. But it seems OS test framework is still using old version. Will remove this change if the conflict is gone.

```
net.bytebuddy:byte-buddy:1.14.7 -> 1.14.9
\--- org.opensearch.test:framework:3.0.0-SNAPSHOT:20240315.134538-2101
     \--- testCompileClasspath
```

File: build.gradle [hunk 2]
```diff
@@ -117,6 +117,7 @@ allprojects {
     configurations.all {
         resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib:1.9.10"
         resolutionStrategy.force "org.jetbrains.kotlin:kotlin-stdlib-common:1.9.10"
+        resolutionStrategy.force "net.bytebuddy:byte-buddy:1.14.9"
```
Comment (dai-chen) [inline build.gradle:120]: OS core bumps Jackson version which in turn bumps byte-buddy version to 1.14.9. But it seems OS test framework is still using old version. Will remove this change if the conflict is gone.

```
net.bytebuddy:byte-buddy:1.14.7 -> 1.14.9
\--- org.opensearch.test:framework:3.0.0-SNAPSHOT:20240315.134538-2101
     \--- testCompileClasspath
```

**Comment by dai-chen** [comment]
Will address conflicts once https://github.com/opensearch-project/sql/pull/2554 merged

**Review by penghuo — APPROVED** [review]

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java [hunk 3]
```diff
@@ -127,6 +131,23 @@ private void executeIndexOp(
...
         flintIndexOpAlter.apply(indexMetadata);
         break;
+      case VACUUM:
+        // Try to perform drop operation first
+        FlintIndexOp tryDropOp =
+            new FlintIndexOpDrop(
+                stateStore, dispatchQueryRequest.getDatasource(), emrServerlessClient);
+        try {
+          tryDropOp.apply(indexMetadata);
+        } catch (IllegalStateException e) {
```
Comment (vamsi-amazon) [inline spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java:141]: Do we want to ignore only IllegalStateException or any kind of error?

**Review by vamsi-amazon — APPROVED** [review]

File: spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java (hunk 3 above)
Comment (dai-chen) [reply spark/src/main/java/org/opensearch/sql/spark/dispatcher/IndexDMLHandler.java:141]: I think only the IllegalStateException thrown from validation? If any exception occurred during operation and commit, there is no need to continue.

**Pull request merged**